[Notifier] Add Pusher Notifier Bridge

bitgandtter opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

Add [Pusher](https://pusher.com/) symfony notifier bridge

bitgandtter committed:
feat: add Pusher symfony notifier bridge

bitgandtter committed:
fix: composer.json package name

bitgandtter committed:
fix: LICENSE and CHANGELOG.md

bitgandtter committed:
chore: apply fabbot

bitgandtter committed:
chore: apply fabbot

bitgandtter committed:
chore: apply fabbot

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/CHANGELOG.md:
```diff
@@ -0,0 +1,7 @@
+CHANGELOG
+=========
+
+6.2.0
```
```suggestion
6.3
```

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/.gitignore:
```diff
@@ -0,0 +1,8 @@
+.php-cs-fixer
+.php-cs-fixer.cache
+.phpcs-cache
+phpcs.xml
```
Please use the same file like the other bridges

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/LICENSE:
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2019-2022 Fabien Potencier
```
```suggestion
Copyright (c) 2022 Fabien Potencier
```

bitgandtter committed:
chore: work on PR comments

bitgandtter committed:
chore: drop declare strict types

bitgandtter committed:
chore: small format changes

bitgandtter committed:
fix: add pusher package to main composer.json to fix tests

bitgandtter commented:
@OskarStark do I need to do something else?

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/CHANGELOG.md:
```diff
@@ -0,0 +1,7 @@
+6.3
+-----
+
+ * Added the bridge
```
```suggestion
 * Add the bridge
```

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/PusherTransportFactory.php:
```diff
@@ -0,0 +1,51 @@
+/**
+ * @author Yasmany Cubela Medina <yasmanycm@gmail.com>
+ */
+#[Autoconfigure(tags: ['texter.transport_factory'])]
```
Not sure about this @fabpot

fabpot replied:
It won't work, right?

bitgandtter replied:
it works on our current project. we use texter since it's the most similar of all available. not sure if you want to introduce something new. I'm all ears

bitgandtter replied:
how should i proceed @OskarStark @fabpot ?

stof replied:
Core services are defined explicitly rather than being autoconfigured (to have more control over what the configuration is). You should define the service with the proper tags in FrameworkBundle instead.

bitgandtter replied:
done @stof

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/README.md:
````diff
@@ -0,0 +1,32 @@
+PUSHER_DSN=pusher://APP_KEY:APP_SECRET@APP_ID?server=SERVER
+```
+
+where:
+
+- `APP_KEY` is your app unique key
+- `APP_SECRET` is your app unique and secret password
+- `APP_ID` is your app unique id
+- `SERVER` is your app server
````
```suggestion
where:

 - `APP_KEY` is your app unique key
 - `APP_SECRET` is your app unique and secret password
 - `APP_ID` is your app unique id
 - `SERVER` is your app server
```

stof replied:
this does not make sense to me. Markdown lists don't require indenting the bullet markers (and if you indent them too much, you end up creating a code block once you reach the minimum indentation of a code block)

OskarStark replied:
IIRC there was a PR in the past by @nicolas-grekas 

I agree with the codeblock, but one space should be no problem

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/README.md:
````diff
@@ -0,0 +1,32 @@
+PUSHER_DSN=pusher://asdasdasd@asdasdasd?server=invalid-server
+PUSHER_DSN=pusher://:asdasdasd@asdasdasd?server=invalid-server
+PUSHER_DSN=pusher://asdadasdasd:asdasdasd@asdasdasd?server=invalid-server
+```
````
Please add the footer like in the other (bridges) README files

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/Tests/PusherOptionsTest.php:
```diff
@@ -0,0 +1,52 @@
+/**
+ * @author Yasmany Cubela Medina <yasmanycm@gmail.com>
+ *
+ * @internal
```
```suggestion
```

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/Tests/PusherTransportFactoryTest.php:
```diff
@@ -0,0 +1,63 @@
+/**
+ * @author Yasmany Cubela Medina <yasmanycm@gmail.com>
+ *
+ * @internal
```
```suggestion
```

OskarStark commented on src/Symfony/Component/Notifier/Bridge/Pusher/Tests/PusherTransportTest.php:
```diff
@@ -0,0 +1,70 @@
+/**
+ * @author Yasmany Cubela Medina <yasmanycm@gmail.com>
+ *
+ * @internal
```
```suggestion
```

bitgandtter committed:
chore: set PusherRecipientInterface as typehint of asPushMessage method
chore: remove @internal annotation from tests
chore: add footer to README.md
chore: apply PR comments

bitgandtter commented:
Hello. from fabpotbot

> in Commits list, a `git rebase` is needed to remove any merges
Merge branch 'symfony:6.2' into pusher-notification
Merge branch 'symfony:6.3' into pusher-notification

excuse my ignorance what do I need to do with it?

thanks in advance

bitgandtter committed:
chore: set RecipientInterface as typehint of asPushMessage method

fabpot commented on src/Symfony/Component/Notifier/Bridge/Pusher/CHANGELOG.md:
```diff
@@ -0,0 +1,7 @@
+=========
+
+6.3
+-----
```
```suggestion
---
```

fabpot replied:
Extra `-`

derrabus committed:
Remove calls to AnnotationRegistry::registerLoader()

derrabus committed:
Remove calls to AnnotationRegistry::registerLoader()

derrabus committed:
Remove calls to AnnotationRegistry::registerLoader()

nicolas-grekas committed:
[PhpUnitBridge] Revert "minor #48725 Remove calls to `AnnotationRegistry::registerLoader()`"

nicolas-grekas committed:
[PhpUnitBridge] Revert "minor #48725 Remove calls to `AnnotationRegistry::registerLoader()`"

fabpot commented:
To fix fabpot, you will need to rebase your PR instead of merging the base branch.
Here is the relevant section of the docs that explain it: https://symfony.com/doc/current/contributing/code/pull_requests.html#rebase-your-pull-request

bitgandtter committed:
chore: fabbot

bitgandtter committed:
fix: LICENSE

nicolas-grekas commented:
Could you please rebase all your PRs and account for https://github.com/symfony/symfony/pull/49385?
We moved all phpunit providers to static methods.

bitgandtter committed:
chore: rebase for #49385

nicolas-grekas commented:
Looks like the rebase didn't go well, sorry.
Please also account for https://github.com/symfony/symfony/pull/49385#issuecomment-1431677676 which I just realized.

bitgandtter commented:
Yes sorry. My bad. Going to fix it soon

bitgandtter committed:
chore: apply comments from #49385

bitgandtter commented:
let me know if this works @nicolas-grekas

bitgandtter committed:
feat: drop autoconfigure tag from transport factory
feat: configure pusher notifier on framework bundle

fabpot commented on src/Symfony/Component/Notifier/Bridge/Pusher/PusherTransport.php:
```diff
@@ -0,0 +1,72 @@
+        $settings = $this->pusherClient->getSettings();
+        preg_match('/api-([\w]+)\.pusher\.com$/m', $settings['host'], $server);
+
+        return sprintf('pusher://%s:%s@%s?server=%s', $settings['auth_key'], $settings['secret'], $settings['app_id'], $server[1]);
```
This method should not return sensitive information like the secret. The goal is not to reconstruct a valid DSN, but to give the main configuration information. See other bridges for some examples.

bitgandtter committed:
chore: drop extra - on CHANGELOG.md
fix: drop sensitive information from toString method on transport

OskarStark commented:
Replaced by
* https://github.com/symfony/symfony/pull/53267
